src/execution_impl.rs

```diff
@@ -13,88 +16,118 @@ use wasmer::{
 use wasmer_as::{Read as ASRead, StringPtr, Write as ASWrite};
 use wasmer_middlewares::Metering;
 
-/// `Print` ABI called by the webassembly VM
-fn print(env: &Env, arg: i32) {
-    let str_ptr = StringPtr::new(arg as u32);
-    println!(
-        "{}",
-        str_ptr
-            .read(env.wasm_env.memory.get_ref().expect("initialized memory"))
-            .unwrap()
-    );
+/// Utility function to call a WASM module (compiled from AssemblyScript) using
+/// high-level Rust types (by de/serializing a String).
+///
+/// The intend is to be able to expose any kind of function as an ABI:
+///
+/// ```ignore
+/// fn get_status() -> Result<NodeStatus> {
+///     Ok(typed_call(env, "0xAddressOfSCInLedger", "get_status", ())?)
+/// }
+/// ```
+pub fn typed_call<T: Serialize, R: DeserializeOwned>(
```

Comment:
I guess, for this time, this whole function and comment could be moved out.

I will re-add it when solving the issue https://github.com/massalabs/sc-runtime/issues/28 my new strategy was now more focused on changing the signature of the `exec` call which is (if I understand well) the only entry point we will expose?

Reply:
@yvan-sraka I remove the function in the next PR if you're ok :) and let's go talking of this in the issue!!